lib/supavisor/db_handler.ex

```diff
@@ -542,10 +544,29 @@ defmodule Supavisor.DbHandler do
   defp handler_caller(%{mode: :session} = data), do: data.caller
   defp handler_caller(_), do: nil
 
-  @spec check_ready(binary()) :: :ready_for_query | :continue
+  @spec check_ready(binary()) ::
+          {:ready_for_query, :idle | :transaction_block | :failed_transaction_block} | :continue
   def check_ready(bin) do
-    if String.ends_with?(bin, Server.ready_for_query()),
-      do: :ready_for_query,
-      else: :continue
+    string_size = byte_size(bin)
+
+    if 6 <= string_size do
+      case bin do
+        <<_::binary-size(string_size - 6), 90, 0, 0, 0, 5, status_indicator::binary>> ->
+          indicator =
+            case status_indicator do
+              <<?I>> -> :idle
+              <<?T>> -> :transaction_block
+              <<?E>> -> :failed_transaction_block
+              _ -> :continue
+            end
+
+          {:ready_for_query, indicator}
+
+        _ ->
+          :continue
+      end
+    else
+      :continue
+    end
```

Comment:
`<<_::binary-size(string_size - 6)>>` is always an error if size is not 6, so I think you can rewrite this to:

```suggestion
    case bin do
      <<90, 0, 0, 0, 5, status_indicator>> ->
          case status_indicator do
            ?I -> :idle
            ?T -> :transaction_block
            ?E -> :failed_transaction_block
            _ -> :continue # maybe this can be removed now?
          end

          {:ready_for_query, indicator}

      _ ->
        :continue
    end
```

WDYT?

Reply:
Ignore me.